lib/Decl.cpp

```diff
@@ -188,4 +195,136 @@ llvm::Value *FunctionDecl::CallFromLiftedBlock(
   }
 }
 
+// Serialize a FunctionDecl to JSON
+llvm::json::Object FunctionDecl::SerializeToJSON() {
+  llvm::json::Object json;
+
+  std::string output_name;
+  if (FLAGS_demangle_names) {
+    output_name = this->demangled_name;
+  } else {
+    output_name = this->name;
+  }
+  json.insert(
+      llvm::json::Object::KV{llvm::json::ObjectKey("name"), output_name});
```

Comment:
This should always be `this->name`, and so if `FLAGS_demangle_names` is true, and if `!this->demangled_name.empty()`, then add in a new key/value pair to the JSON object, `"demangled_name"`.

Reply:
Fixed 028b2a5